src/client/java/io/github/mattidragon/tlaapi/api/plugin/Comparisons.java

```diff
@@ -0,0 +1,45 @@
+package io.github.mattidragon.tlaapi.api.plugin;
+
+import java.util.stream.Stream;
+
+import io.github.mattidragon.tlaapi.api.recipe.TlaStackComparison;
+
+public interface Comparisons<T> {
+    /**
+     * Adds a default comparison method for a stack for its key.
+     * @param item The key for the object to compare.
+     * @param comparison The comparison and hash function
+     */
+    void register(T key, TlaStackComparison comparison);
+
+    /**
+     * Adds a default comparison method for a stack for its key.
+     * @param keys The keys for the object to compare.
+     * @param comparison The comparison and hash function
+     */
+    default void register(TlaStackComparison comparison, @SuppressWarnings("unchecked") T...keys) {
+        for (T key : keys) {
+            register(key, comparison);
+        }
+    }
+
+    /**
+     * Adds a default comparison method for a stack for its key.
+     * @param keys The keys for the object to compare.
+     * @param comparison The comparison and hash function
+     */
+    default void register(TlaStackComparison comparison, Iterable<T> keys) {
+        for (T key : keys) {
+            register(key, comparison);
+        }
+    }
+
+    /**
+     * Adds a default comparison method for a stack for its key.
+     * @param keys The keys for the object to compare.
+     * @param comparison The comparison and hash function
+     */
+    default void register(TlaStackComparison comparison, Stream<T> keys) {
```

Comment:
Is this actually ever useful? I'd think it's niche enough that you can just call `toList` and pass it to the overload above.

Reply:
Yeah fair. I thought I would include it because Streams don't implement the Iterable<T> interface (which I find weird). I'll get rid of it